fix: Fix panic on fetching wafv2 without us-east-1

irmatov opened the pull request:
The bug happens when:

* user has a list of regions that doesn't include us-east-1
* wafv2 resources are fetched

The issue is that multiplexer always produces a client that has us-east-1 as a region and cloudfront scope.
When fetching starts, (*Client).Services call is made, which returns nil because there is no client for us-east-1 region.
This fix changes multiplexer to fetch cloudfront resources only when there is us-east-1 region.

Closes #828

roneli reviewed (approved):
I guess this okay seems a little like a hack

bbernays commented on client/multiplexers.go:
```diff
@@ -59,13 +59,15 @@ func ServiceAccountRegionScopeMultiplexer(service string) func(meta schema.Clien
 					meta.Logger().Trace("region is not supported for service", "service", service, "region", region)
 					continue
 				}
+				if region == "us-east-1" {
+					// fetch cloudfront related resources only when us-east-1 is enabled
+					l = append(l, client.withAccountIDRegionAndScope(accountID, "us-east-1", wafv2types.ScopeCloudfront))
+				}
```
Its not possible to disable `us-east-1` it is only possible to disable regions brought online after March 20, 2019:

```
Regions introduced before March 20, 2019 are enabled by default. You can begin creating and managing resources in these Regions immediately. You cannot enable or disable a Region that is enabled by default.
```

irmatov replied:
"selected" is probably a better word. I meant like "enabled for fetching".

bbernays replied:
I would argue that if a customer wants to grab WAF resources that should outweigh their specification of what regions to grab from. This is the same behavior that S3 has

irmatov replied:
Reverted to older (desired) behavior of always fetching cloudfront scope

irmatov committed:
fix: Fix panic on fetching wafv2 without us-east-1

The bug happens when:

* user has a list of regions that doesn't include us-east-1
* wafv2 resources are fetched

The issue is that multiplexer always produces a client that has
us-east-1 as a region and cloudfront scope. When fetching starts,
(*Client).Services call is made, which returns nil because there
is no client for us-east-1 region.
